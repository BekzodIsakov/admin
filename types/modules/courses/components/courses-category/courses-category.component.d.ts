export const CoursesCategory: any;
//# sourceMappingURL=courses-category.component.d.ts.map