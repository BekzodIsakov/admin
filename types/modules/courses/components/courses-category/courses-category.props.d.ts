//# sourceMappingURL=courses-category.props.d.ts.map