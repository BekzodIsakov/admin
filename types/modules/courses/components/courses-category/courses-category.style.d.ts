export const Wrapper: any;
export const Title: any;
export const CardsGrid: any;
//# sourceMappingURL=courses-category.style.d.ts.map