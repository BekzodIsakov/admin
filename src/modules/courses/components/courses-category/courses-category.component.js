import { hoc } from '@utils/hoc';

import { Title, Wrapper, CardsGrid } from './courses-category.style';

const useCoursesCategoryProps = () => {};

export const CoursesCategory = hoc(
  useCoursesCategoryProps,

  ({ title = 'Title', children }) => {
    return (
      <Wrapper>
        <Title role='heading'>{title}</Title>

        <CardsGrid>{children}</CardsGrid>
      </Wrapper>
    );
  },

  'CoursesCategory'
);
