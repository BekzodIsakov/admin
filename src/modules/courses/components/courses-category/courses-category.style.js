import styled from 'styled-components';

export const Wrapper = styled.div`
  padding: 20px 24px;
  background: ${props => props.theme.colors.white};
  border-radius: 4px;

  & + & {
    margin-top: 20px;
  }
`;

export const Title = styled.div`
  margin-bottom: 24px;
  font-weight: bold;
  font-size: 20px;
  line-height: 24px;
  letter-spacing: -0.02em;
  color: #000000;
`;

export const CardsGrid = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-gap: 17px;

  @media (max-width: 1200px) {
    grid-template-columns: 1fr;
  }
`;
