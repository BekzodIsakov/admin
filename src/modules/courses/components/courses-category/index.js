export * from './courses-category.component.js';
