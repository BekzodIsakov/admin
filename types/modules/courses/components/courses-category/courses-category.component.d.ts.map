{"version":3,"file":"courses-category.component.d.ts","sourceRoot":"","sources":["../../../../../src/modules/courses/components/courses-category/courses-category.component.js"],"names":[],"mappings":"AAMA,kCAcE"}