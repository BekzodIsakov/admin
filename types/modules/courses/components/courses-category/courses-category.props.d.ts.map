{"version":3,"file":"courses-category.props.d.ts","sourceRoot":"","sources":["../../../../../src/modules/courses/components/courses-category/courses-category.props.js"],"names":[],"mappings":""}