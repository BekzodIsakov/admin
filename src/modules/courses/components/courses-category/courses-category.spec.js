import { customRender } from '@test-utils/helpers';
import { screen } from '@testing-library/react';

import { CoursesCategory } from './courses-category.component';

const setup = ({ title, children }) => {
  return customRender(<CoursesCategory title={title} children={children} />);
};

test('Course category component', () => {
  const { asFragment } = setup({ title: 'title', children: 'children' });
  expect(asFragment).toMatchSnapshot();
});

test('Course category props test', () => {
  setup({ title: 'title', children: 'children' });
  const courseCategoryTitle = screen.getByRole('heading', { name: 'title' });
  expect(courseCategoryTitle).toBeTruthy();
});
