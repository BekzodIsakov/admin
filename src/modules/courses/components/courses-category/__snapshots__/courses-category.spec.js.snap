// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course category component 1`] = `[Function]`;
